patches/tModLoader/Terraria/ModLoader/ModItem.cs

```diff
@@ -74,6 +74,11 @@ public class ModItem:ModTexturedType
 		/// </summary>
 		public virtual bool CloneNewInstances => false;
 
+		/// <summary>
+		/// The custom damage type of this ModItem. Assign to ModContent.DamageClassType<T>().
+		/// </summary>
+		public virtual ModDamageClass DamageType => null;
+
```

Comment:
This should be a field or a non-virtual property in Item, not ModItem. People will want to, for example, convert back thrown items from ranged back to thrown.